### PR TITLE
Update Taiwan holidays, add l10n support

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Update Republic of China (Taiwan) holidays:
- add l10n support (`en_US`, `th`, `zh_CN`, **`zh_TW`**), now defaults to `zh_TW` instead of `en_US`
- update holiday name to official Taiwan's [Directorate-General of Personnel Administration, Executive Yuan](https://www.dgpa.gov.tw/information?uid=353&pid=10659) standard

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/locale/en_US/LC_MESSAGES/TW.po [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+msgid "中華民國開國紀念日"
+msgstr "Founding Day of the Republic of China"
+
+#. Chinese New Year's Eve.
+msgid "農曆除夕"
+msgstr "Chinese New Year's Eve"
+
+#. Chinese New Year (Spring Festival).
+msgid "春節"
+msgstr "Chinese New Year (Spring Festival)"
```
Comment (KJhellico) [inline holidays/locale/en_US/LC_MESSAGES/TW.po:34]: Maybe just "Chinese New Year" (as on DGPA site)?

**Review by KJhellico — COMMENTED** [review]
Excellent, as usual! 👍

File: holidays/locale/en_US/LC_MESSAGES/TW.po (hunk 1 above)
Comment (PPsyrius) [reply holidays/locale/en_US/LC_MESSAGES/TW.po:34]: Yeah, that might be for the better if we go all the way with standardization - fixed now.

File: holidays/locale/en_US/LC_MESSAGES/TW.po [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Taiwan holidays en_US localization.
+# Authors: PPsyrius <ppsyrius@ppsyrius.dev>, (c) 2023.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Python Holidays 0.40\n"
+"POT-Creation-Date: 2023-11-24 16:16+0700\n"
+"PO-Revision-Date: 2023-12-02 11:31+0700\n"
+"Last-Translator: PPsyrius <ppsyrius@ppsyrius.dev>\n"
+"Language-Team: Python Holidays Translation Team\n"
```
[the remark this answers is not in the document]
Comment (PPsyrius) [reply holidays/locale/en_US/LC_MESSAGES/TW.po:10]: I might have accidentally based that off older .PO files (I think the culprit here is Thailand's) - we may as well go through them and update them later as needed. 👍

**Review by KJhellico — COMMENTED** [review]
LGTM 🇹🇼

**Review by arkid15r — APPROVED** [review]
Nicely done, LGTM!

**Pull request merged**